### PR TITLE
Tools: sim_vehicle.py: Print a more readable exception

### PR DESCRIPTION
**Pull request by patrickelectric**
Signed-off-by: Patrick José Pereira <patrickelectric@gmail.com>

### CODE REVIEW

**Comment by patrickelectric** [comment]
If a command is not available, sim_vehicle will return this exception:
```
Build commands will be stored in build/sitl/compile_commands.json
'build' finished successfully (1.074s)
SIM_VEHICLE: Using defaults from (/mnt/hd/git/blue/ardupilot/Tools/autotest/default_params/sub.parm)
SIM_VEHICLE: Run ArduSub
SIM_VEHICLE: "/mnt/hd/git/blue/ardupilot/Tools/autotest/run_in_terminal_window.sh" "ArduSub" "/mnt/hd/git/blue/ardupilot/build/sitl/bin/ardusub" "-S" "-I0" "--home" "33.810313,-118.393867,0,270" "-w" "--model" "vectored" "--speedup" "1" "--defaults" "/mnt/hd/git/blue/ardupilot/Tools/autotest/default_params/sub.parm"
SIM_VEHICLE: Run MavProxy
SIM_VEHICLE: "mavproxy.py" "--master" "tcp:127.0.0.1:5760" "--sitl" "127.0.0.1:5501" "--out" "127.0.0.1:14550" "--out" "127.0.0.1:14551" "--mav20" "--out" "udp:0.0.0.0:14550" "--out" "udp:0.0.0.0:14666"
[Run MavProxy] An exception has occurred with command: 'mavproxy.py --master tcp:127.0.0.1:5760 --sitl 127.0.0.1:5501 --out 127.0.0.1:14550 --out 127.0.0.1:14551 --mav20 --out udp:0.0.0.0:14550 --out udp:0.0.0.0:14666'
[Errno 2] No such file or directory: 'mavproxy.py': 'mavproxy.py'
SIM_VEHICLE: Killing tasks
RiTW: Starting ArduSub : /mnt/hd/git/blue/ardupilot/build/sitl/bin/ardusub -S -I0 --home 33.810313,-118.393867,0,270 -w --model vectored --speedup 1 --defaults /mnt/hd/git/blue/ardupilot/Tools/autotest/default_params/sub.parm
which: no osascript in (/usr/local/bin:/usr/local/sbin:/usr/bin:/opt/android-ndk:/opt/android-sdk/platform-tools:/opt/android-sdk/tools:/opt/android-sdk/tools/bin:/opt/brlcad/bin:/opt/cuda/bin:/home/patrick/.local/share/flatpak/exports/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/home/patrick/.gem/ruby/2.4.0/bin:/home/patrick/git/jsbsim/src:/home/patrick/git/blue/ardupilot/Tools/autotest)
```

The old return was:
```
Build commands will be stored in build/sitl/compile_commands.json
'build' finished successfully (3.096s)
SIM_VEHICLE: Using defaults from
(/Users/patrick/git/blue/ardupilot/Tools/autotest/default_params/sub.parm)
SIM_VEHICLE: Run ArduSub
SIM_VEHICLE:
"/Users/patrick/git/blue/ardupilot/Tools/autotest/run_in_terminal_window.sh"
"ArduSub" "/Users/patrick/git/blue/ardupilot/build/sitl/bin/ardusub" "-S"
"-I0" "--home" "-35.363261,149.165230,584,353" "--model" "vectored"
"--speedup" "1" "--defaults"
"/Users/patrick/git/blue/ardupilot/Tools/autotest/default_params/sub.parm"
SIM_VEHICLE: Run MavProxy
SIM_VEHICLE: "mavproxy.py" "--master" "tcp:127.0.0.1:5760" "--sitl" "
127.0.0.1:5501" "--out" "127.0.0.1:14550" "--out" "127.0.0.1:14551"
RiTW: Starting ArduSub :
/Users/patrick/git/blue/ardupilot/build/sitl/bin/ardusub -S -I0 --home
-35.363261,149.165230,584,353 --model vectored --speedup 1 --defaults
/Users/patrick/git/blue/ardupilot/Tools/autotest/default_params/sub.parm
Traceback (most recent call last):
  File "/Users/patrick/git/blue/ardupilot/Tools/autotest/sim_vehicle.py",
line 1072, in <module>
    start_mavproxy(cmd_opts, frame_infos)
  File "/Users/patrick/git/blue/ardupilot/Tools/autotest/sim_vehicle.py",
line 701, in start_mavproxy
    run_cmd_blocking("Run MavProxy", cmd, env=env)
  File "/Users/patrick/git/blue/ardupilot/Tools/autotest/sim_vehicle.py",
line 455, in run_cmd_blocking
    p = subprocess.Popen(cmd, **kw)
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py",
line 394, in __init__
    errread, errwrite)
  File "/usr/local/Cellar/python@2/2.7.15_1/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py",
line 1047, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory
SIM_VEHICLE: Killing tasks
RiTW: Window access not found, logging to /tmp/ArduSub.log
/Users/patrick/git/blue/ardupilot/Tools/autotest/run_in_terminal_window.sh:
line 54: 26469 Terminated: 15          "$cmd" $* >&"$filename" < /dev/null
```

File: Tools/autotest/sim_vehicle.py [hunk 1]
```diff
@@ -452,13 +452,18 @@ def progress_cmd(what, cmd):
...
         progress_cmd(what, cmd)
-    p = subprocess.Popen(cmd, **kw)
-    ret = os.waitpid(p.pid, 0)
-    _, sts = ret
-    if check and sts != 0:
-        progress("(%s) exited with code %d" % (what, sts,))
+    try:
+        p = subprocess.Popen(cmd, **kw)
+        ret = os.waitpid(p.pid, 0)
+        _, sts = ret
```
Comment (peterbarker) [inline Tools/autotest/sim_vehicle.py]: From here to the end of the try block should be moved outside the try block.

**Comment by patrickelectric** [comment]
@peterbarker ty for the tips. Done.

**Comment by peterbarker** [comment]
We format the shell command-lines a little better here (for future reference):

https://github.com/ardupilot/ardupilot/blob/master/Tools/autotest/sim_vehicle.py#L452

**Pull request merged**